output/transform/logs.go

```diff
@@ -62,14 +62,22 @@ func upsertRelationReference(refs []*snapshot.RelationReference, databaseIdx int
 	return idx, refs
 }
 
-func transformPostgresQuerySamples(s snapshot.CompactLogSnapshot, r snapshot.CompactSnapshot_BaseRefs, logState state.TransientLogState) (snapshot.CompactLogSnapshot, snapshot.CompactSnapshot_BaseRefs) {
+func transformPostgresQuerySamples(server state.Server, s snapshot.CompactLogSnapshot, r snapshot.CompactSnapshot_BaseRefs, logState state.TransientLogState) (snapshot.CompactLogSnapshot, snapshot.CompactSnapshot_BaseRefs) {
 	for _, sampleIn := range logState.QuerySamples {
 		occurredAt, _ := ptypes.TimestampProto(sampleIn.OccurredAt)
 
-		if sampleIn.Username == "" || sampleIn.Database == "" || sampleIn.Query == "" {
+		if sampleIn.Query == "" {
 			continue
 		}
 
+		if sampleIn.Username == "" {
+			sampleIn.Username = server.Config.GetDbUsername()
+		}
+
+		if sampleIn.Database == "" {
+			sampleIn.Database = server.Config.GetDbName()
+		}
```

Comment:
Should we warn in these situations? I think this is a great idea in general, since it'll allow more configurations and it's likely to be correct, but it *could* be wrong. Should we point that out? (Though I guess the trick would be to point that out once and not for every sample.)

Reply:
As discussed, since that warning wouldn't show during a test run (since we don't really exercise this code in that situation), I would not warn - but rather have a check up or something in-app that tells you about a log_line_prefix thats missing %u and %d.